pyrealm/pmodel/subdaily.py

```diff
@@ -271,15 +271,15 @@ def __init__(
         r"""Realised daily slow responses in :math:`J_{max25}`"""
 
         # Fill the daily realised values onto the subdaily scale
-        subdaily_tk = self.env.tc + self.env.core_const.k_CtoK
+
+        subdaily_tk = fs_scaler.pad_values(self.env.tc + self.env.core_const.k_CtoK)
```

Comment:
I'm not sure `pad_values` should ever be used outside of `FastSlowScaler.get_daily_means`?

Reply:
That had been my question in https://github.com/ImperialCollegeLondon/pyrealm/pull/169#issuecomment-1948559160

Reply:
So the scaler extracts a set of daily values along the time axis (given a daily window via a `set` method). The modifications for this PR need to allow the the correct values to be extracted when the datetimes don't wrap neatly onto days.

Going back to the subdaily timescale is I think handled automatically by the `scipy` interpolation - we interpolate from the daily values back to the actual provided datetimes so there is no need to modify the behaviour there.

I wasn't entirely sure I wasn't talking rubbish here, so I've pushed an alternative branch with an implementation: https://github.com/ImperialCollegeLondon/pyrealm/tree/72-alternate-implementation

The main thing I wanted to check here is that padding the `be_vie_data_components` passed the correlation test - the new test `test_FSPModel_corr_padded` on this branch sticks 6 hour chunks at the start, end and both for the datetimes on that test and it works as expected. The GPP predictions shouldn't change at all because the data are padded with `np.nan`, so the real calculations start in the same place, but the function accepts the incomplete days without complaint.

Reply:
I forgot to push the actual commits to that branch - now done.

Reply:
Hm, it still tells me the branches are identical?

Reply:
Gah - pushed `develop` instead. Now done!

Reply:
OK, I removed the padding from the subdaily -- I just still don't know why in this case it makes a difference, as the padding should be zero. :-/

Reply:
The padding was also somehow necessary when I did the profiling test, I thought, so checking this. now that I removed it.

Reply:
Hu, it no longer complains in that case. It must have been resolved through previous changes.